stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/entity/QueryChannelsEntity.kt

```diff
@@ -1,26 +1,27 @@
 package io.getstream.chat.android.livedata.entity
 
+import androidx.room.Embedded
 import androidx.room.Entity
 import androidx.room.PrimaryKey
-import io.getstream.chat.android.client.api.models.QuerySort
+import androidx.room.Relation
 import io.getstream.chat.android.client.utils.FilterObject
-import java.util.Date
-import java.util.Objects
 
 @Entity(tableName = "stream_channel_query")
-internal data class QueryChannelsEntity(var filter: FilterObject, val sort: QuerySort) {
+internal data class QueryChannelsEntity(
     @PrimaryKey
-    var id: String
+    var id: String,
+    val filter: FilterObject,
+    val cids: List<String>
+)
 
-    init {
-        // ugly hack to cleanup the filter object to prevent issues with filter object equality
-        filter = FilterObject(filter.toMap())
-        id = (Objects.hash(filter.toMap()) + Objects.hash(sort.data)).toString()
-    }
-
-    var channelCids: List<String> = listOf()
-
-    /** we track when the query was created and updated so we can clear out old results */
-    var createdAt: Date? = null
-    var updatedAt: Date? = null
+@Entity(tableName = "channel_sort_inner_entity")
+internal data class ChannelSortInnerEntity(val name: String, val direction: Int, val queryId: String) {
+    @PrimaryKey
+    var id: String = name.hashCode().toString() + direction.hashCode().toString() + queryId
 }
+
+internal data class QueryChannelsWithSorts(
+    @Embedded val query: QueryChannelsEntity,
+    @Relation(parentColumn = "id", entityColumn = "queryId", entity = ChannelSortInnerEntity::class)
```

Comment:
Does room use a second query or a join for the @relation syntax?

Reply:
Why not define this relation on the QueryChannelsEntity class?

Reply:
You're saying this part is a lot faster than the old approach right? Any idea why?

Reply:
Using relation you mark to room that it should generate two queries when fetching entity. You must create an additional class with relation and it must be not an entity but a simple class with @Relation field. It's Room requirement. You're able free not using this mechanism by create two simple entities and fetch them via two DAO and two queries and build them manually. This Room mechanism does the same but automatically generates all inner routine stuff.

Reply:
I didn't measure it before. But previously QueryChannelsEntity stored QuerySort as json string using serialization via GSON. It think it's not so fast task, SQL queries to DB work should work faster.